### ISSUE TITLE
Mesh Constructor Refactor

### ISSUE THREAD

**Issue by JoshuaSBrown** [parthenon-hpc-lab/parthenon#253]
The mesh constructor has grown to a size that makes it hard to follow. This constructor needs to be broken up into smaller more readable chunks.

**Comment by Yurlungur** [comment]
While I agree the mesh constructor is hard to follow, I'm not sure breaking it up into readable chunks will help. I already find the mesh infrastructure hard to read because the logic jumps around so much through sub-pieces of functionality. The refactor really needs to disentangle the mesh infrastructure.
